carddrawer/src/androidTest/java/com/meli/android/carddrawer/model/CardDrawerViewTest.kt

```diff
@@ -0,0 +1,143 @@
+package com.meli.android.carddrawer.model
+
+import android.content.Context
+import android.graphics.Color
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import com.meli.android.carddrawer.R
+import com.meli.android.carddrawer.ViewScenarioRule
+import com.meli.android.carddrawer.configuration.DefaultCardConfiguration
+import io.mockk.every
+import io.mockk.mockk
+import org.hamcrest.Matchers.endsWith
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class CardDrawerViewTest {
+
+    @get:Rule
+    val viewScenario = ViewScenarioRule()
+
+    private lateinit var subject: CardDrawerView
+    private val expectedCardName = "Maria Alessandra Louise Figueiredo"
+    private val expectedCardExpiration = "12/12"
+    private val cardNumber = "5325  4816  4243  7098"
+    private val expectedCardNumber = "5325  4816  4243  7098"
+    private val cvv = "11"
+    private val expectedCVV = "11**"
+
+    @Before
+    fun setUp() {
+        viewScenario.launch(
+            viewFactory = { CardDrawerView(it) },
+            onView = { subject = it }
+        )
+    }
+
+    @Test
+    fun when_is_rendered_with_tag_then_display_tag_with_expected_values() {
+
+        val expectedTagText = "Sample Tag"
+        val cardDrawerSource = mockk<PaymentCard>(relaxed = true)
+        val tag = buildTag(expectedTagText, "regular")
+
+        every { cardDrawerSource.cardUI } returns buildConfiguration(viewScenario.activity)
+        every { cardDrawerSource.tag } returns tag
+        viewScenario.activity.runOnUiThread { subject.show(cardDrawerSource) }
+
+        onView(withText(expectedTagText.toUpperCase())).check(matches(withId(R.id.card_tag)))
+    }
+
+    @Test
+    fun when_is_rendered_with_bottom_label_then_display_tag_with_expected_values() {
```

Comment:
remover os "_" nos nomes dos tests

Reply:
For some reason having space or special characters in test function names is not allowed in instrumented tests.